### PR TITLE
Initial snapshot tests

### PR DESCRIPTION
**Pull request by RDIL**
These tests are not complete.  This is just the addition of some basic ones which can hopefully be expanded upon in the future.

## Issues
* Right now, most of the components have relay logic.  This really doesnt work because the snapshots just capture the loading.

Also updates dependencies, node to 14 in prod, and added tests to CI

### CODE REVIEW

File: src/components/__tests__/__snapshots__/CirrusCircularProgress.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (RDIL) [inline src/components/__tests__/__snapshots__/CirrusCircularProgress.test.tsx.snap:1]: NOTE: Yes these files are supposed to be committed

File: src/components/__tests__/Logs.test.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+I am a log
+I log things
+`;
+
+  const tree = renderer
+    .create(
+      <Router>
+        <Switch>
+          <Route exact path="/">
+            <Logs logs={exampleLog} />
```
Comment (fkorotkov) [inline src/components/__tests__/Logs.test.tsx:18]: Can we just ignore Router/Switch/Route elements and just test logs?

Comment (RDIL) [reply src/components/__tests__/Logs.test.tsx:18]: Because it has navigation functionality in the component it requires this to work.

**Comment by fkorotkov** [comment]
I was thinking about having screenshot testing originally which seems have couple of benifits:

- easier for us humans to diff
- don't care about underlying structure and focused on visual representation

I'll try to investigate https://github.com/fwouts/react-screenshot-test later this week and if it will work without issues I thing it will make sense to switch (merge your change and switch since you already added tests). @RDIL, what do you think?

**Comment by RDIL** [comment]
Well the issue I noticed here is that because relay doesn't display information right away, the tests are only going to capture the progress bars.  If we had a way to mock relay data, we could implement a lot more.  Also, for that specific library, we would need to eject to modify the jest config.

**Pull request merged**